src/Microsoft.ApplicationInsights.AspNetCore/Implementation/TelemetryConfigurationOptions.cs

```diff
@@ -1,23 +1,59 @@
 ﻿namespace Microsoft.Extensions.DependencyInjection
 {
     using System.Collections.Generic;
+    using System.Linq;
     using Microsoft.ApplicationInsights.Extensibility;
     using Microsoft.Extensions.Options;
 
     /// <summary>
-    /// The <see cref="IOptions{TelemetryConfiguration}"/> implementation that uses <see cref="TelemetryConfiguration.Active"/> as initial value.
+    /// The <see cref="IOptions{TelemetryConfiguration}"/> implementation that create new <see cref="TelemetryConfiguration"/> every time when called".
     /// </summary>
     internal class TelemetryConfigurationOptions : IOptions<TelemetryConfiguration>
     {
         public TelemetryConfigurationOptions(IEnumerable<IConfigureOptions<TelemetryConfiguration>> configureOptions)
         {
-            foreach (var c in configureOptions)
+            // workaround for Microsoft/ApplicationInsights-dotnet#613
+            this.Value = TelemetryConfiguration.CreateDefault();
+
+            var configureOptionsArray = configureOptions.ToArray();
+            foreach (var c in configureOptionsArray)
             {
                 c.Configure(this.Value);
             }
+
+            // as we expect some customers to use TelemetryConfiguration.Active together with dependency injection
+            // we make sure it has been set up, it must be done only once even if there are multiple Web Hosts in the process
```

Comment:
If couple `TelemetryConfigurationOptions` objects can be constructed in parallel, then we may still try to configure `Active` twice due to positive concurrency in the code. Do we expect such behavior or do we need to protect against it all? (I feel like I'm imagining things here..)

Reply:
it's very unlikely, but still possible, so it's a valid concern. I'll add some sync code around. This happens normally once per web host during service resolution, so should not be a performance issue